src/lavinmq/amqp/queue/queue.cr

```diff
@@ -190,13 +190,39 @@ module LavinMQ::AMQP
         @paused.set(true)
       end
       @msg_store = init_msg_store(@data_dir)
+      @empty = @msg_store.empty
+      start
+    end
+
+    private def start
       if @msg_store.closed
         close
+      else
+        handle_arguments
+        spawn queue_expire_loop, name: "Queue#queue_expire_loop #{@vhost.name}/#{@name}" if @expires
+        spawn message_expire_loop, name: "Queue#message_expire_loop #{@vhost.name}/#{@name}"
       end
+    end
+
+    def restart!
```

Comment:
Do we know how the state ended up in closed state? Is there any thing to be aware of before trying to restart a queue or if this fails to restart a queue, how will the user know?

Reply:
We can have an idea on why it's closed based on the log. It should most of the times be because of some kind of exception. We could maybe show that error somewhere?

I think it should be pretty safe to try to restart, but in many cases the queue probably won't start unless one has fixed the reason for why the queue was closed in the first place.